components/brave_wallet_ui/components/shared/add-custom-token-form/add-custom-token-form.tsx

```diff
@@ -0,0 +1,348 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import {
+  AdvancedButton,
+  AdvancedIcon,
+  ButtonRow,
+  DividerRow,
+  DividerText,
+  ErrorText,
+  FormColumn,
+  FormRow,
+  FormWrapper,
+  Input,
+  InputLabel,
+  SubDivider
+} from './add-custom-token-form-styles'
+import { getLocale } from '$web-common/locale'
+import { SelectNetworkDropdown } from '../../desktop'
+import {
+  BraveWallet,
+  WalletState
+} from '../../../constants/types'
+import { NavButton } from '../../extension'
+import { useSelector } from 'react-redux'
+import Amount from '../../../utils/amount'
+import {
+  useAssetManagement,
+  useLib,
+  useTokenInfo
+} from '../../../common/hooks'
+
+interface Props {
+  contractAddress: string | undefined
+  onHideForm: () => void
+}
+
+export const AddCustomTokenForm = (props: Props) => {
+  const {
+    contractAddress,
+    onHideForm
+  } = props
+
+  // state
+  const [hasError, setHasError] = React.useState<boolean>(false)
+  const [showTokenIDRequired, setShowTokenIDRequired] = React.useState<boolean>(false)
+  const [showAdvancedFields, setShowAdvancedFields] = React.useState<boolean>(false)
+  const [showNetworkDropDown, setShowNetworkDropDown] = React.useState<boolean>(false)
+
+  // Form States
+  const [tokenName, setTokenName] = React.useState<string>('')
+  const [tokenID, setTokenID] = React.useState<string>('')
+  const [tokenSymbol, setTokenSymbol] = React.useState<string>('')
+  const [tokenContractAddress, setTokenContractAddress] = React.useState<string>(contractAddress || '')
+  const [tokenDecimals, setTokenDecimals] = React.useState<string>('')
+  const [coingeckoID, setCoingeckoID] = React.useState<string>('')
+  const [iconURL, setIconURL] = React.useState<string>('')
+  const [customAssetsNetwork, setCustomAssetsNetwork] = React.useState<BraveWallet.NetworkInfo>()
+
+  // redux
+  const userVisibleTokensInfo = useSelector(({ wallet }: { wallet: WalletState }) => wallet.userVisibleTokensInfo)
+  const fullTokenList = useSelector(({ wallet }: { wallet: WalletState }) => wallet.fullTokenList)
+  const selectedNetwork = useSelector(({ wallet }: { wallet: WalletState }) => wallet.selectedNetwork)
+  const addUserAssetError = useSelector(({ wallet }: { wallet: WalletState }) => wallet.addUserAssetError)
+
+  // custom hooks
+  const { getBlockchainTokenInfo } = useLib()
+  const {
+    onFindTokenInfoByContractAddress,
+    foundTokenInfoByContractAddress
+  } = useTokenInfo(getBlockchainTokenInfo, userVisibleTokensInfo, fullTokenList, selectedNetwork)
+  const {
+    onAddCustomAsset
+  } = useAssetManagement()
+
+  // Handle Form Input Changes
+  const handleTokenNameChanged = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    if (hasError) {
+      setHasError(false)
+    }
+    setTokenName(event.target.value)
+  }, [hasError])
```

Comment:
you are going to update state/re-render anyways, might just want to always clear the hasError state so you don't need deps in the array